### PR TITLE
Remove Qt 4 support from SConscript.

### PR DESCRIPTION
**Pull request by rryan**

### CODE REVIEW

**Comment by rryan** [comment]
Any objections?

File: build/depends.py [hunk 1]
```diff
@@ -185,27 +180,19 @@ class Qt(Dependence):
...
         import sys
         if sys.platform.startswith('linux'):
             if any(os.access(os.path.join(path, 'pkg-config'), os.X_OK) for path in os.environ["PATH"].split(os.pathsep)):
                 import subprocess
                 try:
-                    if qt5:
-                        qtcore = "Qt5Core"
-                    else:
-                        qtcore = "QtCore"
+                    qtcore = "Qt5Core"
```
Comment (Be-ing) [inline build/depends.py]: This variable is no longer necessary. `qtcore` on the next line can simply be replaced with `Qt5Core`.

File: build/depends.py [hunk 2]
```diff
@@ -275,9 +265,9 @@ def configure(self, build, conf):
         else:
             build.env.Append(CPPDEFINES='QT_SHARED')
 
-        if qt5:
-            # Enable qt4 support.
-            build.env.Append(CPPDEFINES='QT_DISABLE_DEPRECATED_BEFORE')
+        # Enable qt4 deprecated methods.
+        # TODO(rryan): Remove this.
+        build.env.Append(CPPDEFINES='QT_DISABLE_DEPRECATED_BEFORE')
```
Comment (Be-ing) [inline build/depends.py]: Can this be removed now?

File: build/depends.py [hunk 3]
```diff
@@ -290,19 +280,14 @@ def configure(self, build, conf):
 
         # Enable Qt include paths
         if build.platform_is_linux:
-            if qt5 and not conf.CheckForPKG('Qt5Core', '5.0'):
+            if not conf.CheckForPKG('Qt5Core', '5.0'):
                 raise Exception('Qt >= 5.0 not found')
```
Comment (Be-ing) [inline build/depends.py]: What's our actually required minimum version now? I think at least 5.6 or 5.5?

**Review by Be-ing — CHANGES_REQUESTED** [review]
Thanks for working on this cleanup. :)

File: build/depends.py (hunk 2 above)
Comment (rryan) [reply build/depends.py]: Nope, it requires some changes in Mixxx to remove which I started in #1806, but need to take care of @daschuer's comments.

File: build/depends.py (hunk 3 above)
Comment (rryan) [reply build/depends.py]: Hm, I'm not sure. I believe we still build successfully on Trusty which uses [Qt 5.2](https://packages.ubuntu.com/trusty/libs/qt5-default).

File: build/depends.py (hunk 1 above)
Comment (rryan) [reply build/depends.py]: done

**Comment by Be-ing** [comment]
LGTM :+1:

**Pull request merged**